File: Sorts & Search & Shuffling Algorithms/Elementry Search/Interpolation Search/Interpolation.cpp
#include <vector>
#include <cstdlib>
#include <ctime>
using namespace std;
class Interpolation{
public:
    static int Search(vector<int> it, int low, int high, int key){
        int pos;
        if (low <= high && key >= it[low] && key <= it[high]) {
            pos = low
                  + (((double)(high - low) / (it[high] - it[low]))
                     * (key - it[low]));
            if (it[pos] == key)
                return pos;
            if (it[pos] < key)
                return Interpolation::Search(it, pos + 1, high, key);
            if (it[pos] > key)
                return Interpolation::Search(it, low, pos - 1, key);
        }
        return -1;
    }
};

File: Sorts & Search & Shuffling Algorithms/Elementry Search/Interpolation Search/Interpolation.py
class Interpolation:
    @staticmethod
    def Search(iterable, low, high, key):
        if low <= high and iterable[low] <= key <= iterable[high]:
            pos = low + ((high - low) // (iterable[high] - iterable[low]) *
                         (key - iterable[low]))
            if iterable[pos] == key:
                return pos
            if iterable[pos] < key:
                return Interpolation.Search(iterable, pos + 1, high, key)
            if iterable[pos] > key:
                return Interpolation.Search(iterable, low, pos - 1, key)
        return -1